### ISSUE TITLE
4.4.1 Breaks the rest of the dashboard.

### ISSUE THREAD

**Issue by hennaboy** [PrestaShopCorp/ps_mbo#542]
After updating to 4.4.1 the only page that works is the module manager page all other pages of the dashboard now have

PrestaShop\Module\Mbo\Service\View\InstalledModule::__construct(): Argument #4 ($version) must be of type string, float given, called in /home/mydomain.co.uk/public_html/modules/ps_mbo/src/Service/View/ContextBuilder.php on line 276

**Comment by intraordinaire** [comment]
Hello @hennaboy,

Thanks a lot for your report.

That's a strange issue as this parameter is normally well a string nor null.
https://github.com/PrestaShopCorp/ps_mbo/blob/d60998b8f6e6a7c64fd1f73ed2121f0b4762160b/src/Service/View/ContextBuilder.php#L265

And 
https://github.com/PrestaShopCorp/ps_mbo/blob/d60998b8f6e6a7c64fd1f73ed2121f0b4762160b/src/Service/View/ContextBuilder.php#L272
This call should return a string, this is why I don't understand the problem.

Before calling
https://github.com/PrestaShopCorp/ps_mbo/blob/d60998b8f6e6a7c64fd1f73ed2121f0b4762160b/src/Service/View/ContextBuilder.php#L276

We will add a test to be sure before making the construct call, and force to null if the var is not a string.

In the meantime, can you try to clear everything in the cache folder (should be `/home/mydomain.co.uk/public_html/var/cache/`) and try again ?
Also, if the problem persists, can you give me your PrestaShop and PHP versions, so we can try to reproduce.

Thanks !

**Comment by intraordinaire** [comment]
@hennaboy , seems similar to the #541 
Did you have more data in your stacktrace ?

Thanks !

**Comment by hennaboy** [comment]
I resolved the error by changing
$moduleVersion = $module->get('version');

to

$moduleVersion = '4.4.1';

On the assumption of the get('version') is looking for the version number of this module. 

I had been deleting the cache contents whilst looking into getting this resolved. So, cache is not a problem.

PHP8.0 and the error was on Prestashop 8.0.3 and I also upgraded to 8.0.4 to see if there was any fix there but the problem remained. I have also tried PHP8.1 with the same result.

Stack Trace

TypeError:
PrestaShop\Module\Mbo\Service\View\InstalledModule::__construct(): Argument #4 ($version) must be of type string, float given, called in /home/mydomain.co.uk/public_html/modules/ps_mbo/src/Service/View/ContextBuilder.php on line 277

  at modules/ps_mbo/src/Service/View/InstalledModule.php:50
  at PrestaShop\Module\Mbo\Service\View\InstalledModule->__construct()
     (modules/ps_mbo/src/Service/View/ContextBuilder.php:277)
  at PrestaShop\Module\Mbo\Service\View\ContextBuilder->getInstalledModules()
     (modules/ps_mbo/src/Service/View/ContextBuilder.php:176)
  at PrestaShop\Module\Mbo\Service\View\ContextBuilder->getCommonContextContent()
     (modules/ps_mbo/src/Service/View/ContextBuilder.php:95)
  at PrestaShop\Module\Mbo\Service\View\ContextBuilder->getViewContext()
     (modules/ps_mbo/src/Traits/Hooks/UseDashboardZoneOne.php:40)
  at ps_mbo->hookDashboardZoneOne()
     (classes/Hook.php:983)
  at HookCore::coreCallHook()
     (classes/Hook.php:409)
  at HookCore::callHookOn()
     (classes/Hook.php:918)
  at HookCore::exec()
     (controllers/admin/AdminDashboardController.php:297)
  at AdminDashboardControllerCore->renderView()
     (classes/controller/AdminController.php:2236)
  at AdminControllerCore->initContent()
     (classes/controller/Controller.php:319)
  at ControllerCore->run()
     (classes/Dispatcher.php:510)
  at DispatcherCore->dispatch()
     (admin6495ihdga/index.php:92)     

This may also be useful to you

Uncaught Error: PrestaShop\Module\Mbo\Service\View\InstalledModule::__construct(): Argument #4 ($version) must be of type string, float given, called in /home/hennaboy.co.uk/public_html/modules/ps_mbo/src/Service/View/ContextBuilder.php on line 277
{
    "exception": {}
}

and

Uncaught PHP Exception TypeError: "PrestaShop\Module\Mbo\Service\View\InstalledModule::__construct(): Argument #4 ($version) must be of type string, float given, called in /home/hennaboy.co.uk/public_html/modules/ps_mbo/src/Service/View/ContextBuilder.php on line 277" at /home/hennaboy.co.uk/public_html/modules/ps_mbo/src/Service/View/InstalledModule.php line 50
{
    "exception": {}
}

I did read issue 541 but I could not see anything in my stacktrace that would point to another module although that of course could be the issue it just has not presented itself.

**Comment by intraordinaire** [comment]
@hennaboy , oh indeed my bad, argument 4 is module version and not module URL. Not well awake this morning.
This solve your issue indeed, but the 4.4.1 is for the MBO module. Here, we want to check the version of another module.
So I assume a module is having something like this

```php
public function __construct()
    {
        $this->name = 'module_name';
        $this->version = 1.0;
```

instead of
```php
public function __construct()
    {
        $this->name = 'module_name';
        $this->version = '1.0';
```

That may cause this issue.

We will force the cast of the version to avoid this error :
```php
$installedModules[] = (new InstalledModule($moduleId, $moduleName, $moduleStatus, (string) $moduleVersion, $moduleConfigUrl))->toArray();
```

Can you try it on line 276 of the ContextBuilder ?

And, if you find the module with version like this (float instead of string), can you tell us, so we can try to better handle this kind of problem ? (that seems to be the case for the `statsgeolocation` module responsible for #541 )
Thanks !

**Comment by intraordinaire** [comment]
Great !
We'll publish a release with this fix soon.

Thanks a lot for your report, and don't hesitate in case of problem !

**Comment by intraordinaire** [comment]
Fixed in 4.4.2.

Thanks for the report !
